src/mixins/PageData.ts

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable max-classes-per-file */
+import { Component, Vue } from 'vue-property-decorator';
+import { Route, RouteCallback } from 'vue-router';
+
+/**
+ * Abstraction for handling fetching data for a page while allowing it to be cached
+ * if provided by SSR
+ *
+ * Here be abstract class mixin hackery
+ * https://github.com/vuejs/vue-class-component/issues/91#issuecomment-451422547
+ */
+// @ts-ignore
+@Component
+abstract class PageDataMixinAbstract extends Vue {
+  abstract async fetchPageData(): Promise<any>;
```

Comment:
What good does it do to mark this as abstract if you derive from it immediately? Doesn't that revoke the `abstract` value?
(If not, wouldn't that break having to mark the class as `abstract` to have abstract functions?)

Reply:
You would think. :P Hence the "Here be abstract class mixin hackery" - read about it here: https://github.com/vuejs/vue-class-component/issues/91#issuecomment-451422547

Reply:
OH - I didn't realize the @ts-ignore did that.
I'll look into it, hopefully it's possible to fix it via changes to TS that we can issue (assuming they didn't already)